split volume module into alsa and pulseaudio

erizocosmico opened the pull request:
Fixes #87 

This makes it possible to use the pulseaudio package without having
the ALSA headers installed.

soumya92 commented:
Overall this looks good. There is an increase in things that are accessible outside the volume package though. And with these being separate package, it's staring to be similar to the weather module (one overall module, with multiple "providers" for the data source). Can you make a few changes?

- Add a `func MakeVolume(min, max, volume int32, controller Controller) Volume` so that the Volume instance passed to format and click handlers does not expose the controller directly. (Since the controller impls don't perform the bounds checks and state updates that Volume.SetVolume/SetMuted do).

- volume.CreateModule can just be volume.New, much like weather.New that takes in a weather Provider. Along the same lines, rename `ModuleImpl` to `Provider`.

- Return volume.Provider from the alsa/pulseaudio functions, and update call sites to be `volume.New(alsa.DefaultMixer())`, or `volume.New(pulseaudio.DefaultSink())`.

Since this is a breaking change for the module, I would prefer the API to be final before I merge this in. Thanks!

erizocosmico commented:
@soumya92 All requested changes applied! I changed the min, max and vol to int64 to honor the types in the struct and added mute as well to avoid having to set it with `vol.Mute = mute` later instead of during value creation.

soumya92 commented:
This is great! Thanks!
